internal/databases/sqlserver/sqlserver.go

```diff
@@ -551,3 +552,30 @@ func RunOrReuseProxy(ctx context.Context, cancel context.CancelFunc, folder stor
 	}
 	return &LockWrapper{lock}, nil
 }
+
+func GetDBRestoreLSN(db *sql.DB, databaseName string) (int64, error) {
+	query := fmt.Sprintf(`SELECT MAX(redo_start_lsn) 
+							FROM sys.master_files
+							WHERE database_id=DB_ID('%s') 
+							`, databaseName)
```

Comment:
It is looks like a potential SQL Injections.
I would propose:
`
query := "SELECT MAX(redo_start_lsn)  FROM sys.master_files WHERE database_id=DB_ID(?)"
db.QueryRow(query, databaseName).Scan(...)
`